mod_regression/controllers.py

```diff
@@ -114,11 +114,27 @@ def test_result(regression_id):
     pass
 
 
-@mod_regression.route('/test/new')
+@mod_regression.route('/test/new', methods=['GET', 'POST'])
+@template_renderer()
+@check_access_rights([Role.admin])
 def test_add():
-    # Add a new regression test
-    pass
-
+    """
+    Function to add a regression test
+    """
+    form = AddTestForm(request.form)
+    form.sample_id.choices = [(sam.id,sam.sha) for sam in Sample.query.all()]
```

Comment:
Please make sure the PR is PEP-8 compliant... I see a missing space between the entries in the tuple ;)